flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/functions/utils/AggSqlFunction.scala

```diff
@@ -107,7 +107,8 @@ object AggSqlFunction {
   private[flink] def createOperandTypeInference(
       name: String,
       aggregateFunction: UserDefinedAggregateFunction[_, _],
-      typeFactory: FlinkTypeFactory): SqlOperandTypeInference = {
+      typeFactory: FlinkTypeFactory,
+      externalAccType: DataType): SqlOperandTypeInference = {
```

Comment:
NIT: Use Scala Option? We can just:
```
val actualSignature = accType.map(_.getLogicalType).orNull +: operandTypeInfo
```
This is simple and better readability?

Reply:
Isn't the `externalAccType` always non-null? It is used for `accumulate` method.

Reply:
@wuchong You are right, it should be non-null.

Reply:
Thank you for your advice. So, can we just write as follow?
`val actualSignature = externalAccType.getLogicalType +: operandTypeInfo`

Reply:
I think so. Could you rename `operandTypeInfo` to `operandLogicalType` ?

Reply:
Yes, it will be better to rename `operandTypeInfo` to `operandLogicalType`

Reply:
@wuchong @JingsongLi  I have modified it, i am honored that you can review it again